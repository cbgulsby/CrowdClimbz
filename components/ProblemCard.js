import React from 'react'
import { View, Text, TouchableOpacity, StyleSheet, Dimensions } from 'react-native';

const screenWidth = Dimensions.get('window').width - 10;

export default function TempProblemCard(props) {
    return (
        <TouchableOpacity style={styles.container} onPress = {() => {
            props.problemInfo.cardNavigation.navigate("ProblemCardScreen", {
                betaVideo: props.problemInfo.betaVideo,
                cardNavigation: props.problemInfo.cardNavigation,
                date: props.problemInfo.date,
                description: props.problemInfo.description,
                grade: props.problemInfo.grade,
                gymName: props.problemInfo.gymName,
                inappropriateFlag: props.problemInfo.inappropriateFlag,
                outOfDateFlag: props.problemInfo.outOfDateFlag,
                problemName: props.problemInfo.problemName,
                photo: props.problemInfo.photo,
                time: props.problemInfo.time,
                user: props.problemInfo.user,
            })}}>
            <View style={styles.leftInnerContainer}>
                <View style={styles.gymNameContainer}>
                    <Text 
                        style={{fontSize: 30, color: '#073B4C'}}>
                        {props.problemInfo.problemName}
                    </Text>
                </View>
                <View style={styles.gymLocationContainer}>
                    <Text 
                        style={{fontSize: 15, color: '#073B4C'}}>
                        {props.problemInfo.gymName}
                    </Text>
                </View>
            </View>
            <View style={styles.rightInnerContainer}>
                <Text 
                    style={{fontSize: 40, color: '#073B4C'}}>
                    V{props.problemInfo.grade}
                </Text>
            </View>
        </TouchableOpacity>
    );
}

const styles = StyleSheet.create({
    container: {
        flexDirection: 'row', 
        backgroundColor: '#06D6A0', 
        borderWidth: 2, 
        borderColor: 'black', 
        margin: 3, 
        height: 100,
        width: screenWidth,
        alignSelf: 'center',
        borderRadius: 5
    },
    leftInnerContainer: {
        flex: 3
    },
    rightInnerContainer: {
        flex: 1,
        justifyContent: 'center', 
        alignItems: 'center'
    },
    gymNameContainer: {
        flex: 1,
        justifyContent: 'flex-end', 
        paddingLeft: 5
    },
    gymLocationContainer: {
        flex: 1, 
        justifyContent: 'flex-start', 
        paddingLeft: 5
    }
})