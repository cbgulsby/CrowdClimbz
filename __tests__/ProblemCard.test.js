import React from 'react';
import renderer from 'react-test-renderer';

import ProblemCard from '../components/ProblemCard';

// it('works', () => {
//     expect(1).toBe(1);
//   });


test('renders correctly', () => {
  const tree = renderer.create(<ProblemCard />).toJSON();
  expect(tree).toMatchSnapshot();
});