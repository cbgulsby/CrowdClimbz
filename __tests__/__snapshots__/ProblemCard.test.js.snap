// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  accessible={true}
  focusable={false}
  isTVSelectable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": "#2576f7",
      "borderColor": "black",
      "borderWidth": 1,
      "flexDirection": "row",
      "height": 100,
      "margin": 3,
      "opacity": 1,
    }
  }
>
  <View
    style={
      Object {
        "flex": 3,
      }
    }
  >
    <View
      style={
        Object {
          "flex": 1,
          "justifyContent": "flex-end",
          "paddingLeft": 5,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "white",
            "fontSize": 30,
          }
        }
      />
    </View>
    <View
      style={
        Object {
          "flex": 1,
          "justifyContent": "flex-start",
          "paddingLeft": 5,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "white",
            "fontSize": 15,
          }
        }
      />
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "white",
          "fontSize": 40,
        }
      }
    >
      V
    </Text>
  </View>
</View>
`;
